src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,101 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    public void setup() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_success_Ok() {
+        User actual;
+        try {
+            actual = registrationService.register(user);
+        } catch (ValidationException e) {
+            return;
+        }
+        assertEquals(actual.getLogin(), storageDao.get(actual.getLogin()));
+    }
+
+    @Test
+    void register_anExisting_Login_Not_Ok() {
+        Storage.people.add(user);
+        try {
+            registrationService.register(user);
+        } catch (ValidationException e) {
+            return;
+        }
+        fail("Exception must be throw");
```

Comment:
try/catch very rarely used and/or required in tests.
Please avoid it and use assertThrows structure for all tests